tests/stubs/devices/mockserver.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devices
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/paypal/gatt"
+	"github.com/paypal/gatt/examples/option"
+
+	"github.com/kubeedge/kubeedge/tests/stubs/devices/services"
+)
+
+func createServiceAndAdvertise(d gatt.Device, s gatt.State) {
+	// Setup GAP and GATT services for Linux implementation.
+	// OS X doesn't export the access of these services.
+	d.AddService(services.NewGapService("sensortagmock")) // no effect on OS X
+	d.AddService(services.NewGattService())               // no effect on OS X
+
+	// Creating a temperature reading service
+	temperatureSvc := services.NewTemperatureService()
+	d.AddService(temperatureSvc)
+
+	// Advertise device name and service's UUIDs.
+	log.Print("Advertising device name and service UUID")
+	d.AdvertiseNameAndServices("sensortagmock", []gatt.UUID{temperatureSvc.UUID()})
+
+	// Advertise as an OpenBeacon iBeacon
+	log.Print("Advertise as an OpenBeacon iBeacon")
+	d.AdvertiseIBeacon(gatt.MustParseUUID("AA6062F098CA42118EC4193EB73CCEB6"), 1, 2, -59)
+}
+
+func main() {
+	d, err := gatt.NewDevice(option.DefaultServerOptions...)
+	if err != nil {
+		log.Fatalf("Failed to open device, err: %s", err)
```

Comment:
Should we exit here ?

Reply:
It will fail if it is not able to create a NewDevice. This mainly happens when it cant get hci. So none of  the operations associated with device can be performed. So I guess exit is fine..Thoughts please!!!!